Assets/Scripts/Missions/MissionKitchen1.cs

```diff
@@ -280,10 +280,12 @@ private void OnFadingCompleteRestartMission()
             }
 
             GameManager.Instance.GetPlayerTransform().position = denturesCheckpointRespawnPosition;
+            GameManager.Instance.GetMovementController().ResetVelocity();
         }
         else
         {
             GameManager.Instance.GetPlayerTransform().position = respawnPosition;
+            GameManager.Instance.GetMovementController().ResetVelocity();
```

Comment:
Can't we just remove lines 283 and 288 to outside the if-else conditionals or does it need to be reset before you do the `FadeIn`?

Reply:
True! I think resetting it before / after doesn't have a noticeable effect at all, so should be good!